File: python/Convert Sorted Array to Binary Search Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def buildBSTNode(self, nums):
        if (len(nums) == 1):
            return TreeNode(nums[0])
        elif (len(nums) == 0):
            return None
        else:
            mid = (0 + len(nums)) / 2
            root = TreeNode(nums[mid])
            root.left = self.buildBSTNode(nums[0:mid])
            root.right = self.buildBSTNode(nums[mid + 1:len(nums)])
        return root

    def sortedArrayToBST(self, nums):
        return self.buildBSTNode(nums)

File: cpp/Convert Sorted Array to Binary Search Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> vals;
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        vals = nums;
        return helper(0, nums.size() - 1);
    }
    
    TreeNode* helper(int low, int high) {
        if (low > high) {
            return NULL;
        } else if (low == high) {
            TreeNode *node = new TreeNode(vals[low]);
            return node;
        }
        
        int mid = (low + high) / 2;
        TreeNode *root = new TreeNode(vals[mid]);
        root->left = helper(low, mid - 1);
        root->right = helper(mid + 1, high);
        
        return root;
    }
};
